internal/controller/user.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"github.com/avialog/backend/internal/dto"
+	"github.com/avialog/backend/internal/service"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type UserController interface {
+	GetProfile(*gin.Context)
+	UpdateProfile(*gin.Context)
+}
+
+type userController struct {
+	userService service.UserService
+}
+
+func newUserController(userService service.UserService) UserController {
+	return &userController{userService: userService}
+}
+
+func (u *userController) GetProfile(c *gin.Context) {
+	// there will be getting user id from JWT token
+	userID := uint(1)
+	// -----------------------
+	user, err := u.userService.GetProfile(userID)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()}) // or should I return status not found (404) ?
```

Comment:
First of all, never return `http.StatusInternalServerError` for logic err - it is reserved for actions that failed by internal reasons (for example database connection failed).
Secondly, if user was actually not found, you should return 404. You should be able to do it with `errors.As` or `errors.Is`.

However, as our application has special mechanism for Firebase Auth, we are sure, that user exists (our future middleware actually will do upsert). So due to that, leave this code as it is. But remember, that this is not always the case.